src/data/common/D2ApiDataStore.ts

```diff
@@ -1,32 +1,28 @@
-import _ from "lodash";
 import { D2Api } from "../../types/d2-api";
-import { DataSource, isDhisInstance } from "../../domain/instance/entities/DataSource";
 import { getD2APiFromInstance } from "../../utils/d2-utils";
 import { DataStore, DataStoreKey } from "../../domain/metadata/entities/MetadataEntities";
 import { promiseMap } from "../../utils/common";
 import { DataStoreMetadata } from "../../domain/data-store/DataStoreMetadata";
+import { Instance } from "../../domain/instance/entities/Instance";
 
 export class D2ApiDataStore {
     private api: D2Api;
 
-    constructor(instance: DataSource) {
-        if (!isDhisInstance(instance)) {
-            throw new Error("Invalid instance type for MetadataD2ApiRepository");
-        }
+    constructor(instance: Instance) {
         this.api = getD2APiFromInstance(instance);
     }
 
-    async getDataStore(filter: { namespaces: string[] }): Promise<DataStore[]> {
+    async getDataStore(filter: { namespaces?: string[] }): Promise<DataStore[]> {
         const response = await this.api.request<string[]>({ method: "get", url: "/dataStore" }).getData();
-        const namespacesWithKeys = await this.getAllKeysFromNamespaces(response);
-        return filter.namespaces.length === 0
-            ? namespacesWithKeys
-            : _(namespacesWithKeys)
-                  .map(namespace => {
-                      return filter.namespaces.includes(namespace.id) ? namespace : undefined;
+        const namespacesWithKeys = await this.getAllKeysFromNamespaces(
+            filter.namespaces
+                ? response
+                : DataStoreMetadata.getDataStoreIds(filter.namespaces || []).map(ns => {
```

Comment:
Isn't the condition reversed? if you have `filter.namespaces`, then you can call `getDataStoreIds`.